crates/telio-nurse/src/heartbeat.rs

```diff
@@ -954,15 +998,200 @@ impl Analytics {
             _ => false,
         }
     }
+
+    async fn fetch_conn_data(
+        &self,
+        hb_info: &mut HeartbeatInfo,
+        disconnect: bool,
+        index_map: &HashMap<PublicKey, u8>,
+    ) -> Result<(), Box<dyn error::Error>> {
+        if disconnect {
+            self.aggregator.force_save_unacknowledged_segments().await;
+        }
+
+        let AggregatorCollectedSegments { relay, peer } =
+            self.aggregator.collect_unacknowledged_segments().await;
+
+        let (mut peers_str, mut relay_str) = (String::new(), String::new());
+        let index_us = *index_map.get(&self.public_key).ok_or(io::Error::new(
+            io::ErrorKind::NotFound,
+            "Our index not found",
+        ))?;
+
+        for p_segment in peer.iter() {
+            let mut writer = WriterBuilder::new()
+                .flexible(true)
+                .delimiter(b':')
+                .terminator(csv::Terminator::Any(b','))
+                .from_writer(vec![]);
+            let index_other = *index_map.get(&p_segment.node).ok_or(io::Error::new(
+                io::ErrorKind::NotFound,
+                "Node's index not found",
+            ))?;
+
+            peers_str.push_str(&index_us.to_string());
+            peers_str.push(':');
+            peers_str.push_str(&index_other.to_string());
+            peers_str.push(':');
+            peers_str.push_str(&p_segment.duration.as_secs().to_string());
+            peers_str.push(':');
```

Comment:
Hmm, couldn't you use this writer to serialize this data as well? It already has this delimiter set, so it feels like it could be nicer

Reply:
Addressed